### PR TITLE
Track remote host/IP for blocked points

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/ChannelByteArrayHandler.java [hunk 1]
```diff
@@ -75,6 +76,7 @@ protected void channelRead0(ChannelHandlerContext ctx, byte[] msg) throws Except
       if (rootCause != null && rootCause.getMessage() != null) {
         errMsg = errMsg + ", root cause: \"" + rootCause.getMessage() + "\"";
       }
+      errMsg += "; remote: " + ((InetSocketAddress)ctx.channel().remoteAddress()).getHostString();
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/ChannelByteArrayHandler.java]: you need a space after `(InetSocketAddress)` for perfection =p

File: proxy/src/main/java/com/wavefront/agent/ChannelByteArrayHandler.java [hunk 2]
```diff
@@ -92,6 +94,7 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
     if (rootCause != null && rootCause.getMessage() != null) {
       message += ", root cause: \"" + rootCause.getMessage() + "\"";
     }
+    message += "; remote: " + ((InetSocketAddress)ctx.channel().remoteAddress()).getHostString();
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/ChannelByteArrayHandler.java]: ditto (it is formatted correctly below)

File: proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java [hunk 3]
```diff
@@ -108,6 +110,9 @@ public static void processPointLine(final String message,
       if (rootCause != null && rootCause.getMessage() != null) {
         errMsg = errMsg + ", root cause: \"" + rootCause.getMessage() + "\"";
       }
+      if (ctx != null) {
+        errMsg += "; remote: " + ((InetSocketAddress) ctx.channel().remoteAddress()).getHostString();
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java]: remoteaddress can technically be null

File: proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java [hunk 4]
```diff
@@ -125,6 +130,7 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
     if (rootCause != null && rootCause.getMessage() != null) {
       message += ", root cause: \"" + rootCause.getMessage() + "\"";
     }
+    message += "; remote: " + ((InetSocketAddress) ctx.channel().remoteAddress()).getHostString();
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java]: remoteAddress can technically be null

File: proxy/src/main/java/com/wavefront/agent/OpenTSDBPortUnificationHandler.java [hunk 5]
```diff
@@ -291,9 +292,16 @@ private void writeHttpResponse(HttpRequest request,
...
+   * @param ctx       ChannelHandlerContext (optional) to extract remote client ip
    */
-  private void blockMessage(final String messageId, final String message, @Nullable final Throwable e) {
+  private void blockMessage(final String messageId,
+                            final String message,
+                            @Nullable final Throwable e,
+                            @Nullable final ChannelHandlerContext ctx) {
     String errMsg = messageId + ": OpenTSDB: " + message;
+    if (ctx != null) {
+      errMsg = errMsg + "; remote: " + ((InetSocketAddress)ctx.channel().remoteAddress()).getHostString();
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/OpenTSDBPortUnificationHandler.java]: ditto for formatting and nullability for remoteAddress()

**Comment by basilisk487** [comment]
@panghy Done!

**Pull request merged**